### PR TITLE
Add FXIOS-9749-10153 new Edit bookmark view and Edit folder view

### PR DESCRIPTION
**Pull request by FilippoZazzeroni**
### Edit bookmark view part 1
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9749)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/21418)

### Edit bookmark folder
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10153)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/22249)

## :bulb: Description
- Added feature flag `bookmarkRefactorFeature` and created new file for bookmarks panel `BookmarksViewController`. 
- Old code has been renamed to legacy.
- Added Edit Bookmark module and Edit Folder module.
- Removed options to create new bookmark from panel context menu in `BookmarksViewController`.

https://github.com/user-attachments/assets/64b35732-b636-4fe9-97f9-4d503f686000

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: BrowserKit/Sources/ComponentLibrary/DefaultTextField.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Common
+import UIKit
+
+public class DefaultTextField: UITextField, ThemeApplicable {
```
Comment (cyndichin) [inline BrowserKit/Sources/ComponentLibrary/DefaultTextField.swift:9]: thanks for adding a new component to the library! Could we add this to the sample component library as well as update the docs for this new reusable component? Can be done in a separate PR, but wanted to make sure we are tracking this.

File: firefox-ios/Client/Frontend/Library/Bookmarks/Edit Folder/EditFolderViewController.swift [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Life Cycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        _ = UIImage(resource: .addToHomescreenLarge)
```
Comment (dataports) [inline firefox-ios/Client/Frontend/Library/Bookmarks/Edit Folder/EditFolderViewController.swift:66]: Is this line still needed?

File: firefox-ios/Client/Frontend/Library/Bookmarks/Edit Folder/EditFolderViewController.swift [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let section = Section(rawValue: indexPath.section) else { return }
+        if section == .parentFolder, let folder = viewModel.folderStructures[safe: indexPath.row] {
+            viewModel.selectFolder(folder)
+        }
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        guard let sectionEnum = Section(rawValue: section), sectionEnum == .parentFolder else { return nil }
+        guard let header = tableView.dequeueReusableHeaderFooterView(withIdentifier: UX.parentFolderHeaderIdentifier)
```
Comment (dataports) [inline firefox-ios/Client/Frontend/Library/Bookmarks/Edit Folder/EditFolderViewController.swift:196]: could these two guards be combined?

File: firefox-ios/Client/Frontend/Library/Bookmarks/Utitlity/BookmarksSaver.swift [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    let profile: Profile
+
+    func save(bookmark: FxBookmarkNode, parentFolderGUID: String) async -> Result<Void, Error> {
+        return await withCheckedContinuation { continuation in
+            let operation: Success? = {
+                switch bookmark.type {
+                case .bookmark:
+                    // no parent create it
```
Comment (dataports) [inline firefox-ios/Client/Frontend/Library/Bookmarks/Utitlity/BookmarksSaver.swift:26]: I'm not sure what this comment means, does it mean the bookmark doesn't have a folder parent?

File: firefox-ios/Client/Frontend/Library/Bookmarks/Edit Folder/EditFolderViewController.swift (hunk 3 above)
Comment (FilippoZazzeroni) [reply firefox-ios/Client/Frontend/Library/Bookmarks/Edit Folder/EditFolderViewController.swift:196]: Yes sure, i'm going to combine those

File: firefox-ios/Client/Frontend/Library/Bookmarks/Edit Folder/EditFolderViewController.swift (hunk 2 above)
Comment (FilippoZazzeroni) [reply firefox-ios/Client/Frontend/Library/Bookmarks/Edit Folder/EditFolderViewController.swift:66]: No sorry, i forgot it there

File: BrowserKit/Sources/ComponentLibrary/DefaultTextField.swift (hunk 1 above)
Comment (FilippoZazzeroni) [reply BrowserKit/Sources/ComponentLibrary/DefaultTextField.swift:9]: Yes Cyndi. i'll add a task to create docs for it and add it to the sample app

File: firefox-ios/Client/Frontend/Library/Bookmarks/Edit Folder/EditFolderViewController.swift (hunk 2 above)
Comment (FilippoZazzeroni) [reply firefox-ios/Client/Frontend/Library/Bookmarks/Edit Folder/EditFolderViewController.swift:66]: Sorry i forgot it there, i'm going to remove it.

File: firefox-ios/Client/Frontend/Library/Bookmarks/Utitlity/BookmarksSaver.swift (hunk 4 above)
Comment (FilippoZazzeroni) [reply firefox-ios/Client/Frontend/Library/Bookmarks/Utitlity/BookmarksSaver.swift]: Yes. I removed it since it can be also understand by the code.

**Comment by FilippoZazzeroni** [comment]
Thanks very much for the review @dataports @cyndichin 😃 i've updated the code as per code review

**Review by dataports — APPROVED** [review]
LGTM, thanks for your work on this! 🚀

**Pull request merged**